tests/suites/test_suite_psa_crypto_slot_management.function

```diff
@@ -643,12 +643,21 @@ void invalid_handle( )
     TEST_ASSERT( handle1 != 0 );
 
     /* Attempt to close and destroy some invalid handles. */
-    TEST_EQUAL( psa_close_key( 0 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_close_key( handle1 - 1 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_close_key( handle1 + 1 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_destroy_key( 0 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_destroy_key( handle1 - 1 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_destroy_key( handle1 + 1 ), PSA_ERROR_INVALID_HANDLE );
+    if( handle1 - 1 != 0 )
+    {
+        TEST_EQUAL( psa_close_key( handle1 - 1 ), PSA_ERROR_INVALID_HANDLE );
+        TEST_EQUAL( psa_destroy_key( handle1 - 1 ), PSA_ERROR_INVALID_HANDLE );
+    }
+    if( handle1 + 1 != 0 )
```

Comment:
When would handle1 + 1 be equal to zero? I don't think `psa_import_key()` would ever return -1 for a handle, would it? Certainly not in this test. What are we trying to test with this and how can we guarantee we test it?

Reply:
It won't happen with our implementation (`handle1` is deterministically `PSA_KEY_SLOT_COUNT`), but `(psa_key_handle_t)(-1)` is a potentially valid handle.

The goal of this test is to close a nonzero, invalid handle and check that this seems to keep the key store in a good state. We check this by checking that `handle1` is still there and that the shutdown is clean.